Do not return entitlements error

smarlowucf opened the pull request:
In the case of an error getting entitlements log the error and add the message to the batch failure list. This will allow the message to be processed later.

smarlowucf committed:
Do not return entitlements error

In the case of an error getting entitlements log the error and
add the message to the batch failure list. This will allow the
message to be processed later.

schaefi reviewed (approved):
yes makes sense :+1:

smarlowucf committed:
Raise exception if get entitlements error

This is caught and logged and the message id is flagged as failed.
This allows the message to be saved and reprocessed.

smarlowucf commented:
Updated the PR to raise an exception which is caught and handled by the existing try/except block. This prevents code duplication.
